src/main/java/com/aws/greengrass/lifecyclemanager/UpdateSystemPolicyService.java

```diff
@@ -96,20 +94,21 @@ public Set<String> getPendingActions() {
     @SuppressWarnings("PMD.AvoidCatchingThrowable")
     protected void runUpdateActions(String deploymentId) {
         try (LockScope ls = LockScope.lock(lock)) {
-            for (Map.Entry<String, UpdateAction> todo : pendingActions.entrySet()) {
-                try {
-                    actionInProgress.set(todo.getKey());
-                    todo.getValue().getAction().run();
-                    logger.atDebug().setEventType("service-update-action").addKeyValue("action", todo.getKey()).log();
-                } catch (Throwable t) {
-                    logger.atError().setEventType("service-update-action-error").addKeyValue("action", todo.getKey())
-                            .setCause(t).log();
-                }
+            final UpdateAction pendingUpdateAction = pendingActions.remove(deploymentId);
```

Comment:
- Pending actions that only correspond to the specified deployment id should be performed. 
- Rely on  `pendingActions` map to check if the update action is in progress/completed or never added.

Reply:
Does this mean that a deferred deployment cannot be cancelled? Ideally, if a component is deferring a config merge due a deployment, this deployment should still be cancellable given that defer period is still ongoing.

Reply:
- Deferred deployments are still cancellable. That code path is not modified. 
- `runUpdateActions` is run only after we waited enough for the deferred time.
- While we're waiting for the deferred time, deployment pending actions can be discarded upon request.